File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    participant.sort()
    completion.sort()

    #point_com=0
    #point_par=0    
    #for i in range(point_par,len(participant)) :
     #   for k in range(point_com,len(completion)) :        
      #      if participant[i] ==completion[k] :
       #         participant[i] = '*'
        #        point_com+=1
          #      point_par+=1
           # else :
            #    point_par+=1
    j=0       
    for i in range(0,len(participant)) :
        if j<len(completion) and participant[i] == completion[j] :
            participant[i] = '*'
            completion[j] = '*'
            j+=1

        else :
            pass

        
    for j in participant :
        if j == '*' :
            pass
        else :
            answer+=j
            
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.Map;
import java.util.HashMap;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String, Integer> parMap = new HashMap<>();
        
        for (String s : participant){
            if(parMap.containsKey(s)){
                parMap.put(s, parMap.get(s) + 1);
            }
            else{
                parMap.put(s,1);
            }
        }
        
        for( String a : completion){
            if(parMap.containsKey(a)){
                parMap.put(a,parMap.get(a) - 1);
            }
        }
            
        for(String name : parMap.keySet()) {
            if (parMap.get(name) != 0) {
                answer += name;
            }
        }
        
        return answer;
    }
}